packages/ui/src/components/MultiSelectColumnFilter.tsx

```diff
@@ -0,0 +1,185 @@
+import clsx from 'clsx';
+import { type JSX, useMemo, useState } from 'preact/compat';
+import Dropdown from 'react-bootstrap/Dropdown';
+
+type FilterMode = 'any' | 'all';
+
+function computeSelected<T extends readonly any[]>(
+  allValues: T,
+  mode: 'include' | 'exclude',
+  selected: Set<T[number]>,
+) {
+  if (mode === 'include') {
+    return selected;
+  }
+  return new Set(allValues.filter((s) => !selected.has(s)));
+}
+
+function defaultRenderValueLabel<T>({ value }: { value: T }) {
+  return <span>{String(value)}</span>;
+}
+
+/**
+ * A component that allows the user to filter a column containing arrays of values.
+ * Supports two matching modes:
+ * - 'any': Show rows where ANY of the selected values are present (OR logic)
+ * - 'all': Show rows where ALL of the selected values are present (AND logic)
+ *
+ * @param params
+ * @param params.columnId - The ID of the column
+ * @param params.columnLabel - The label of the column, e.g. "Rubric Items"
+ * @param params.allColumnValues - All possible values that can appear in the column
+ * @param params.renderValueLabel - A function that renders the label for a value
+ * @param params.columnValuesFilter - The current state of the column filter
+ * @param params.setColumnValuesFilter - A function that sets the state of the column filter
+ * @param params.matchMode - Whether to match ANY or ALL selected values
+ * @param params.setMatchMode - A function to change the match mode
+ */
+export function MultiSelectColumnFilter<T extends readonly any[]>({
+  columnId,
+  columnLabel,
+  allColumnValues,
+  renderValueLabel = defaultRenderValueLabel,
+  columnValuesFilter,
+  setColumnValuesFilter,
+  matchMode = 'any',
+  setMatchMode,
+}: {
+  columnId: string;
+  columnLabel: string;
+  allColumnValues: T;
+  renderValueLabel?: (props: { value: T[number]; isSelected: boolean }) => JSX.Element;
+  columnValuesFilter: T[number][];
+  setColumnValuesFilter: (value: T[number][]) => void;
+  matchMode?: FilterMode;
+  setMatchMode?: (mode: FilterMode) => void;
+}) {
+  const [includeExcludeMode, setIncludeExcludeMode] = useState<'include' | 'exclude'>('include');
+
+  const selected = useMemo(
+    () => computeSelected(allColumnValues, includeExcludeMode, new Set(columnValuesFilter)),
+    [includeExcludeMode, columnValuesFilter, allColumnValues],
+  );
+
+  const apply = (newMode: 'include' | 'exclude', newSelected: Set<T[number]>) => {
```

Comment:
It seems like we want a 'matches any' vs 'matches all' mode for this filter. This include/exclude goes away.